Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/mtl_state_cache.mm

```diff
@@ -943,6 +943,60 @@ void ToObjC(const RenderPassStencilAttachmentDesc &desc,
     return re.first->second;
 }
 
+static bool ValidateRenderPipelineState(const MTLRenderPipelineDescriptor *descriptor,
+                                        ContextMtl *context,
+                                        const mtl::ContextDevice &device)
+{
+    // Ensure there is at least one valid render target.
+    bool hasValidRenderTarget = false;
+
+    for (NSUInteger i = 0;
+         !hasValidRenderTarget && i < GetMaxNumberOfRenderTargetsForDevice(device); ++i)
+    {
+        auto colorAttachment = descriptor.colorAttachments[i];
+        if (colorAttachment && colorAttachment.pixelFormat != MTLPixelFormatInvalid)
+        {
+            hasValidRenderTarget = true;
+        }
+    }
+
+    if (!hasValidRenderTarget && descriptor.depthAttachmentPixelFormat != MTLPixelFormatInvalid)
+    {
+        hasValidRenderTarget = true;
+    }
+
+    if (!hasValidRenderTarget && descriptor.stencilAttachmentPixelFormat != MTLPixelFormatInvalid)
+    {
+        hasValidRenderTarget = true;
+    }
+
+    if (!hasValidRenderTarget)
+    {
+        ANGLE_MTL_HANDLE_ERROR(context, "Render pipeline requires at least one render target.",
```

Comment:
It would increase our chances to get error reporting for this case in webkit bugs if we could get the errors through the GL error extension to show up somehow in the inspector..

Reply:
Is it possible to annotate calls with errors in the inspector? At a minimum, it would be beneficial to log these errors to the web inspector console.